python/cudf/cudf/core/indexed_frame.py

```diff
@@ -1694,6 +1694,193 @@ def last(self, offset):
             slice_func=lambda i: self.iloc[i:],
         )
 
+    @annotate("SAMPLE", color="orange", domain="cudf_python")
+    def sample(
+        self,
+        n=None,
+        frac=None,
+        replace=False,
+        weights=None,
+        random_state=None,
+        axis=None,
+        ignore_index=False,
+    ):
+        """Return a random sample of items from an axis of object.
+
+        If reproducible results are required, a random number generator may be
+        provided via the `random_state` parameter. This function will always
+        produce the same sample given an identical `random_state`.
+
+        Notes
+        -----
+        When sampling from ``axis=0/'index'``, ``random_state`` can be either
+        a numpy random state (``numpy.random.RandomState``) or a cupy random
+        state (``cupy.random.RandomState``). When a numpy random state is
+        used, the output is guaranteed to match the output of the corresponding
+        pandas method call, but generating the sample may be slow. If exact
+        pandas equivalence is not required, using a cupy random state will
+        achieve better performance, especially when sampling large number of
+        items. It's advised to use the matching `ndarray` type to the random
+        state for the `weights` array.
+
+        Parameters
+        ----------
+        n : int, optional
```

Comment:
These docstrings are very inconsistent about how the document optional parameters and their default values. It would be good to clean that up a little if possible.

Reply:
This might be something specific to this API. The default value is unspecified because it's conditional on which keyword argument the user specifies on. Similar things like: https://github.com/rapidsai/cudf/blob/branch-22.04/python/cudf/cudf/core/tools/datetimes.py, where any 3 out of 4 of the argument can be specified, and the default argument kind of varies as inputs.

Reply:
Yeah I see. Not a big deal then.